Feat: Add replaceBlockAttribute hook for searching & replacing custom fields in custom blocks (#42)
badasswp/search-and-replace: README.md

**Parameters**

- allowedBlocks _`{string[]}`_ List of Allowed Blocks.
  <br/>

> #### `search-replace-for-block-editor.allowedBlocksAndFields`

Reviewer: ```suggestion
#### `search-replace-for-block-editor.replaceBlockAttribute`
```

> This custom hook (filter) provides the ability to include the search and replace functionality for your custom block with custom fields:

Reviewer: ```suggestion
This custom hook (action) provides users the ability to include the search and replace functionality for custom blocks with custom properties:
```

Author: @badasswp readme is adjusted.

```js

> import { addFilter } from '@wordpress/hooks';

Reviewer: ```suggestion
import { addAction } from '@wordpress/hooks';
```

Author: @badasswp readme is adjusted.

> addFilter(

Reviewer: Based on my earlier refactor comment, pls change this to:

```js
addAction(
    'search-replace-for-block-editor.replaceBlockAttribute',
    'yourAttribute',
    ( replaceBlockAttribute, name, args ) => {
        const { element: block } = args;
        if ( block.hasOwnProperty( 'fields' ) ) {
            block.fields.forEach( ( property ) => {
                replaceBlockAttribute( args, property );
            } );
        }
    }
);
```

  'search-replace-for-block-editor.allowedBlocksallowedBlocksAndFields',
  'yourBlock',
  (allowedBlocksAndFields) => {
    allowedBlocksAndFields.push({
      name: 'yourBlock',
      fields: ['text', 'label', 'title'],
    });
    return allowedBlocksAndFields;
  }
);
```

**Parameters**

> - allowedBlocksAndFields _`{object[]}`_ List of Allowed Blocks and it fields. Inside the object the a name property for the name of the block and an array fields for the search fields of this block.

Reviewer: ```suggestion
- replaceBlockAttribute _`{Function}`_ By default, this is a function that takes in an `args` and `property` as params.
- name _`{string}`_ By default, this is a string containing the `name` of the block.
- args _`{Object}`_ By default, this is an object containing the `element`, `pattern`, `text` and `status`.
```

Author: @badasswp readme is adjusted.

**Parameters**

- allowedBlocks _`{string[]}`_ List of Allowed Blocks.

>   <br/>

Reviewer: ```suggestion
<br/>
```

Author: these are fixed

#### `search-replace-for-block-editor.replaceBlockAttribute`

This custom hook (action) provides users the ability to include the search and replace functionality for custom blocks with custom properties:

```js
import { addAction } from '@wordpress/hooks';

addAction(
	'search-replace-for-block-editor.replaceBlockAttribute',
	'yourBlock',
	( replaceBlockAttribute, name, args ) => {
		const { element: block } = args;
		if ( block.hasOwnProperty( 'fields' ) ) {
			block.fields.forEach( ( property ) => {
				replaceBlockAttribute( args, property );
			} );
		}
	}
);
```

**Parameters**

- replaceBlockAttribute _`{Function}`_ By default, this is a function that takes in an `args` and `property` as params.
- name _`{string}`_ By default, this is a string containing the `name` of the block.

> - args _`{Object}`_ By default, this is an object containing the `element`, `pattern`, `text` and `status`.

Reviewer: ```suggestion
- args _`{Object}`_ By default, this is an object containing the `element`, `pattern`, `text` and `status`.
<br/>
```

Author: added  the break.
